docs/agent/config/v3.mdx

```diff
@@ -748,3 +774,162 @@ endpoints:
     upstream:
       url: 80
 ```
+
+### Agent TLS Termination
+
+```bash
+# 1. Generate Root CA private key and certificate
+openssl genpkey -algorithm RSA -out ca.key -pkeyopt rsa_keygen_bits:2048
+openssl req -x509 -new -nodes -key ca.key -sha256 -days 365 -out ca.crt -subj "/CN=ExampleCA"
+
+# 2. Generate server private key, CSR, and signed certificate
+openssl genpkey -algorithm RSA -out server.key -pkeyopt rsa_keygen_bits:2048
+openssl req -new -key server.key -out server.csr -subj "/CN=your-domain.ngrok.io"
+openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -days 365 -sha256
+
+# 3. Generate client private key, CSR, and signed certificate
+openssl genpkey -algorithm RSA -out client.key -pkeyopt rsa_keygen_bits:2048
+openssl req -new -key client.key -out client.csr -subj "/CN=ExampleClient"
+openssl x509 -req -in client.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out client.crt -days 365 -sha256
+
+# 4. Verify certificate validity
+openssl verify -CAfile ca.crt server.crt
+# Expected Output:
+# server.crt: OK
+
+openssl verify -CAfile ca.crt client.crt
+# Expected Output:
+# client.crt: OK
+
+# 5. Inspect certificate contents
+openssl x509 -in ca.crt -text -noout
+openssl x509 -in server.crt -text -noout
+openssl x509 -in client.crt -text -noout
+
+# 6. Create ngrok configuration file
+cat <<EOF > /home/<your-user>/.config/ngrok/ngrok.yml
+version: 3
+agent:
+  authtoken: 1yEmu1I7Nk4SqxKtdQwVbcBGdHk_53NW4yEmNDBnufjb1XDod
+endpoints:
+  - name: foo
+    url: 'tls://your-domain.ngrok.io'
+    upstream:
+      url: 8080
+    agent_tls_termination:
+      server_certificate: /home/<your-user>/.config/ngrok/server.crt
+      server_private_key: /home/<your-user>/.config/ngrok/server.key
+      mutual_tls_certificate_authorities:
+        - /home/<your-user>/.config/ngrok/ca.crt
+EOF
+
+# 7. Start the ngrok endpoint
+ngrok api start --config=/home/<your-user>/.config/ngrok/ngrok.yml --log=stdout --log-level=debug foo
+
+# 8. Verify connection to the endpoint with Root CA only
+curl -v https://your-domain.ngrok.io --cacert ca.crt
+# Expected Output:
+# * CONNECTED(00000003)
+# * depth=1 CN = ExampleCA
+# * verify return:1
+# * depth=0 CN = your-domain.ngrok.io
+# * verify return:1
+# * SSL handshake successful
+# * Verify return code: 0 (ok)
+
+# 9. Verify mutual TLS connection with client certificate
+curl -v https://your-domain.ngrok.io \
+  --cacert ca.crt \
+  --cert client.crt \
+  --key client.key
```

Comment:
cert/key options shouldn't be necessary if you're not testing mTLS yet (which is implied because there is another section later about mTLS?)